modules/tinymce/src/core/main/ts/newline/ReduceNestedFonts.ts

```diff
@@ -0,0 +1,29 @@
+import { Arr, Optional } from '@ephox/katamari';
+import { Attribute, Compare, Css, Remove, SugarElement, Traverse, SugarNode } from '@ephox/sugar';
+
+const reduceFontStyleNesting = (block: Element, node: Element): void => {
+  const blockSugar = SugarElement.fromDom(block);
+  const nodeSugar = SugarElement.fromDom(node);
+  const isEndBlock = (element: SugarElement) => Compare.eq(element, blockSugar);
+  const elements = [ nodeSugar, ...Traverse.parentsElement(nodeSugar, isEndBlock) ];
```

Comment:
If you change this to `PredicateFilter.ancestors(nodeSugar, hasFontSize, isEndBlock)`  and then also do a hasFontSize check on the nodeSugar then we you have a set of elements that you need to process.

Then the first one is the one you keep and the rest if the ones you remove the font size from and unwrap if it doesn't have any attributes anymore. I think we should only unwrap `span` elements since in theory the element could be say `strong` then it has some styling.